.travis.yml

```diff
@@ -0,0 +1,101 @@
+langauge: c
+dist: trusty
+
+compiler:
+  - gcc
+  - clang
+
+matrix:
+  include:
+  - name: "GCC 4.9"
+    addons:
+      apt:
+        sources:
+          - ubuntu-toolchain-r-test
+        packages:
+          - libopenmpi-dev
+          - openmpi-bin
+          - g++-4.9
+    env :
+      - MATRIX_EVAL="CC=gcc-4.9 && CXX=g++-4.9"
+
+  - name: "GCC 6"
+    addons:
+      apt:
+        sources:
+          - ubuntu-toolchain-r-test
+        packages:
+          - libopenmpi-dev
+          - openmpi-bin
+          - g++-6
+    env :
+      - MATRIX_EVAL="CC=gcc-6 && CXX=g++-6"
+
+  - name: "GCC 7"
+    addons:
+      apt:
+        sources:
+          - ubuntu-toolchain-r-test
+        packages:
+          - libopenmpi-dev
+          - openmpi-bin
+          - g++-7
+    env :
+      - MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
+
+  - name: "GCC 8"
+    addons:
+      apt:
+        sources:
+          - ubuntu-toolchain-r-test
+        packages:
+          - libopenmpi-dev
+          - openmpi-bin
+          - g++-8
+    env :
+      - MATRIX_EVAL="CC=gcc-8 && CXX=g++-8"
+
+  - name: "Clang 4.0"
+    addons:
+      apt:
+        sources:
+          - llvm-toolchain-trusty-4.0
+        packages:
+          - libopenmpi-dev
+          - openmpi-bin
+          - clang-4.0
+    env :
+      - MATRIX_EVAL="CC=clang-4.0 && CXX=clang++-4.0"
+
+  - name: "Clang 6.0"
+    addons:
+      apt:
+        sources:
+          - ubuntu-toolchain-r-test
+          - llvm-toolchain-trusty-6.0
+        packages:
+          - libopenmpi-dev
+          - openmpi-bin
+          - clang-6.0
+    env :
+      - MATRIX_EVAL="CC=clang-6.0 && CXX=clang++-6.0"
+
+before_install:
+  - eval "${MATRIX_EVAL}"
+  - git clone https://github.com/ECP-VeloC/KVTree.git $HOME/KVTree
+  - mkdir $HOME/KVTree/kvtree-build
+  - mkdir $HOME/KVTree/kvtree-install
+  - cd $HOME/KVTree/kvtree-build
+  - cmake .. -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_INSTALL_PREFIX=$HOME/KVTree/kvtree-install -DMPI=ON
+  - make
+  - make install
+
+script:
+  - cd $TRAVIS_BUILD_DIR
+  - cmake -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_C_FLAGS="-coverage" -DMPI=ON -DAXL_ASYNC_DAEMON=OFF -DWITH_KVTREE_PREFIX=$HOME/KVTree/kvtree-install
```

Comment:
Hmm, I do have `-DMPI=ON` but
```
CMake Warning:
  Manually-specified variables were not used by the project:

    MPI
```
still shows up. This shows up when I build locally as well, but only for AXL.

Reply:
Yea, I think we can get rid of it